conjure-java-jersey-server/src/main/java/com/palantir/conjure/java/server/jersey/GuavaOptionalParamConverterProvider.java

```diff
@@ -24,10 +24,12 @@
 import javax.ws.rs.ext.ParamConverterProvider;
 import javax.ws.rs.ext.Provider;
 import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.jersey.internal.inject.Custom;
 import org.glassfish.jersey.internal.inject.Providers;
 import org.glassfish.jersey.internal.util.ReflectionHelper;
 import org.glassfish.jersey.internal.util.collection.ClassTypePair;
 
+@Custom
```

Comment:
This annotation ensures that our custom param converters are considered first. 
Otherwise the order in which the param converter providers are considered is arbitrary:
https://github.com/jersey/jersey/blob/2.25.1/core-server/src/main/java/org/glassfish/jersey/server/internal/inject/ParamConverterFactory.java#L58-L101

For example, here is an order that was used when I was debugging tests locally:
![param-converter-providers](https://user-images.githubusercontent.com/5273164/70758299-6548ef80-1cf7-11ea-8b05-b6dc62b8fc6f.png)

This is especially important for the `AuthHeaderParamConverterProvider` and `BearerTokenParamConverterProvider` to ensure that we don't fallback to the built-in `AggregatedProvider` which contains the `TypeValueOf` param converter (which is used currently for `AuthHeader` and `BearerToken`).

Reply:
Something along these lines would make an excellent comment in the code :-)